interfaces/builtin/unity8.go

```diff
@@ -40,6 +128,26 @@ func (iface *Unity8Interface) PermanentPlugSnippet(plug *interfaces.Plug, securi
 }
 
 func (iface *Unity8Interface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		oldTags := []byte("###SLOT_SECURITY_TAGS###")
+		newTags := slotAppLabelExpr(slot)
+		snippet := bytes.Replace([]byte(unity8ConnectedPlugAppArmor), oldTags, newTags, -1)
+
+		// FIXME: Until we decide whether unity8 is going to use Snappy-style
+		//        appIDs (snap.NAME.COMMAND) or Touch-style ones
+		//        (NAME_COMMAND_REVISION), we'll use a simpler *NAME* glob
+		//        here. This glob does not provide isolation because 'foo' can
+		//        access 'barfoobaz'. The base declaration restriction cannot
+		//        be lifted until this is properly mediated.
+		appidsOld := []byte("###PLUG_DBUS_APPIDS###")
+		appidsNew := fmt.Sprintf("*%s*", dbus.SafePath(plug.Snap.Name()))
```

Comment:
In the interests of getting some of this landed, I have taken out the two dbus permissions that dealt with appids (receiving url dispatches and interacting with the unity8 launcher).

We can discuss the appid issue separately.